ykrt/src/compile/jitc_yk/jit_ir.rs

```diff
@@ -89,7 +89,8 @@ pub enum Operand {
 
 impl Operand {
     pub(crate) fn new(kind: OpKind, val: u64) -> Self {
-        if val < 2u64.pow(u32::try_from(SHORT_OPERAND_VALUE_SIZE).unwrap()) {
+        // check if the operand's value can fit in a short operand.
+        if val & (u64::MAX << SHORT_OPERAND_VALUE_SIZE) == 0 {
```

Comment:
It's still not obvious to me what `SIZE` is (OK, OK, I know *now*, but I'll forget later, because it's unclear). Can we put all the maths in the constant at the top of the file when it will become much clearer?

Reply:
Did https://github.com/ykjit/yk/pull/930/commits/eb47831171ace788c850f6fe9b6530fe8d219151 not work?

Reply:
OK the docs on `SIZE` help, but let's still stick all the maths in the constant (if nothing else, it makes clearer a) what the values actually are b) they're compile-time constants).

Reply:
I think that would need to be another PR as there are lots of computations we'd want to move to the top.

Reply:
All of them are introduced in this PR aren't they? Or are they preexisting?

Reply:
OK, promise me you'll fix it then :)